component/discovery/scaleway/scaleway.go

```diff
@@ -0,0 +1,182 @@
+package scaleway
+
+import (
+	"encoding"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/config"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/river/rivertypes"
+	prom_config "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	prom_discovery "github.com/prometheus/prometheus/discovery/scaleway"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"gopkg.in/yaml.v2"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "discovery.scaleway",
+		Args:    Arguments{},
+		Exports: discovery.Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+type Arguments struct {
+	Project         string            `river:"project_id,attr"`
+	Role            Role              `river:"role,attr"`
+	APIURL          string            `river:"api_url,attr,optional"`
+	Zone            string            `river:"zone,attr,optional"`
+	AccessKey       string            `river:"access_key,attr"`
+	SecretKey       rivertypes.Secret `river:"secret_key,attr,optional"`
+	SecretKeyFile   string            `river:"secret_key_file,attr,optional"`
+	NameFilter      string            `river:"name_filter,attr,optional"`
+	TagsFilter      []string          `river:"tags_filter,attr,optional"`
+	RefreshInterval time.Duration     `river:"refresh_interval,attr,optional"`
+	Port            int               `river:"port,attr,optional"`
+
+	ProxyURL        config.URL       `river:"proxy_url,attr,optional"`
+	TLSConfig       config.TLSConfig `river:"tls_config,block,optional"`
+	FollowRedirects bool             `river:"follow_redirects,attr,optional"`
+	EnableHTTP2     bool             `river:"enable_http2,attr,optional"`
```

Comment:
Looks like the upstream configuration is embedding the entire `HTTPClientConfig` object here. Is there a reason we're not doing the same?

Reply:
Yes; I originally embedded the whole HTTPClientConfig, but Prometheus only uses a subset of the fields from HTTPClientConfig, and doesn't document that the others are supported.

For consistency with other discovery components, I follow the same pattern: only expose the subset of fields we actually use and support.